### PR TITLE
Simplify logic in output to handle multiple code and fix ITER cylinder SDDR

### PR DESCRIPTION
**Pull request by dodu94**
# Description

This PR should be reviewed after PR #308 

The PR addresses two issues:
1) the if-else statements across the output.py did not allow to perform post-process of d1s benchmarks. While correcting them, the logic has been simplified a little bit but there may be still optimizations to be done (out of scope of this PR)
2) adress some issues specific of ITER cylinder. After the introduction of F4Enix, a couple of edge cases that where handled by JADE were not transposed into F4Enix and they had to be restored here.

## Type of change

Please select what type of change this is.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New benchmark
    - [ ] Non-breaking change which entirely uses exisiting classes, structure etc
    - [ ] Breaking change which has implemented new/modified classes etc
- [ ] New feature 
    - [ ] Non-breaking change which adds functionality
    - [ ] Breaking change fix or feature that would cause existing functionality to not work as expected

# Testing

Manual testing + CI test added for at least the single pp of ITER cyl. the test is a bit slow but necessary

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] General testing 
    - [x] New and existing unit tests pass locally with my changes
    - [x] Coverage is >80%

### CODE REVIEW

File: tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/d1s_job_script.err
Comment (alexvalentine94) [inline tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/d1s_job_script.err:1]: can remove

File: tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/d1s_job_script.out
Comment (alexvalentine94) [inline tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/d1s_job_script.out:1]: can remove

File: tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/d1s_job_script.err
Comment (alexvalentine94) [inline tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/d1s_job_script.err:1]: can remove

File: tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/d1s_job_script.out
Comment (alexvalentine94) [inline tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/d1s_job_script.out:1]: can remove

File: tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/ITER_Cyl_SDDRo
Comment (alexvalentine94) [inline tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/ITER_Cyl_SDDRo:1]: Do we need this?

File: tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/ITER_Cyl_SDDRo
Comment (alexvalentine94) [inline tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/ITER_Cyl_SDDRo:1]: Do we need this?

File: jade/default_settings/Benchmarks_Configuration/SphereSDDR/irrad_93c
[the remark this answers is not in the document]
Comment (dodu94) [reply jade/default_settings/Benchmarks_Configuration/SphereSDDR/irrad_93c:1]: I opened an issue about this #321

File: tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/d1s_job_script.err
Comment (dodu94) [reply tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/d1s_job_script.err:1]: removed

File: tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/d1s_job_script.out
Comment (dodu94) [reply tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/d1s_job_script.out:1]: Using new F4Enix parsing capabilities JADE looks both in the .out and output file for the name (version of MCNP or D1SUNED) of the code used to be inserted in the metadata files

File: tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/d1s_job_script.err
Comment (dodu94) [reply tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/d1s_job_script.err:1]: removed

File: tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/d1s_job_script.out
Comment (dodu94) [reply tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/d1s_job_script.out:1]: see previous comment

File: tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/ITER_Cyl_SDDRo
Comment (dodu94) [reply tests/TestFiles/output/Simulations/99c/ITER_Cyl_SDDR/d1s/ITER_Cyl_SDDRo:1]: see previous comment

File: tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/ITER_Cyl_SDDRo
Comment (dodu94) [reply tests/TestFiles/output/Simulations/93c/ITER_Cyl_SDDR/d1s/ITER_Cyl_SDDRo:1]: see previous comment

**Review by alexvalentine94 — APPROVED** [review]

**Pull request merged**